### PR TITLE
Add footnote about plugin hooks

### PR DESCRIPTION
**Pull request by DrJosh9000**
Currently, multiple plugin-provided checkout and command hooks can run. This is a long-standing issue (see buildkite/agent#1161) but evidently we didn't document it well.

### CODE REVIEW

File: pages/agent/v3/hooks.md [hunk 1]
```diff
@@ -136,17 +136,19 @@ they are run as part of each job:
...
 | `post-artifact` | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span> | Runs after artifacts have been uploaded, if an artifact upload pattern was defined for the job. |
 | `pre-exit`      | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span> | Runs before the job finishes. Useful for performing cleanup tasks. |
 {: class="table table--no-wrap"}
 
+[^plugin-exception]: For legacy compatibility reasons, this restriction does not apply to plugins. All checkout or command hooks provided by plugins will be run in the order of plugins, which means that multiple checkout hooks and command hooks can run. However, any checkout hooks and command hooks provided by plugins will prevent any repository or agent hooks of the same type from running.
```
Comment (DrJosh9000) [inline pages/agent/v3/hooks.md:150]: It seems this does not render as a footnote. Is there a better way to format this?

Comment (mbelton-buildkite) [reply pages/agent/v3/hooks.md:150]: Interesting! I haven't seen footnotes used very heavily in dev docs, but I wonder if it's worth exploring. From what I can see, our tooling does support this syntax but it's not enabled. For now, I suggest adding a new section after "Creating job lifecycle hooks:"

```md
### Hook exception for plugins

As of Agent v3.15.0, only the first checkout and command hooks will run if there are multiple found. However, there is an exception with plugins for legacy compatibility. All checkout or command hooks provided by plugins will run in the order the plugins are specified, meaning multiple checkout and command hooks can run. Note that checkout hooks and command hooks provided by plugins will prevent any repository or agent hooks of the same type from running.
```

And linking to it in the table like:

```md
..., only the first will be run (see [Hook exception for plugins](#job-lifecycle-hooks-hook-exception-for-plugins))
```

What do you think?

Comment (DrJosh9000) [reply pages/agent/v3/hooks.md]: Sounds good to me.

File: pages/agent/v3/hooks.md [hunk 2]
```diff
@@ -136,17 +136,21 @@ they are run as part of each job:
...
 | `pre-checkout`  | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span>                                                                                                                    | Runs before checkout. |
-| `checkout`      | <span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-agent">Agent</span>                                                                                                                    | Overrides the default git checkout behavior.<br>*Note:* As of Agent v3.15.0, if multiple checkout hooks are found, only the first will be run. |
+| `checkout`      | <span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-agent">Agent</span>                                                                                                                    | Overrides the default git checkout behavior.<br>*Note:* As of Agent v3.15.0, if multiple checkout hooks are found, only the first will be run (see [Hook exception for plugins](#job-lifecycle-hooks-hook-exception-for-plugins)). |
```
[the remark this answers is not in the document]
Comment (DrJosh9000) [reply pages/agent/v3/hooks.md]: Done - wdyt?

[the remark this answers is not in the document]
Comment (triarius) [reply pages/agent/v3/hooks.md]: Nice, LGTM!

**Review by triarius — APPROVED** [review]
LGTM 🎉

**Review by mbelton-buildkite — APPROVED** [review]
Looks good to me!

**Comment by mbelton-buildkite** [comment]
The build is failing due to the URL checker finding a flaky link. I've removed that link so merging `main` should solve any issues with that.

**Pull request merged**